spec/Action/CaptureActionSpec.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+
+namespace spec\BitBag\SyliusImojePlugin\Action;
+
+use BitBag\SyliusImojePlugin\Api\ImojeApi;
+use BitBag\SyliusImojePlugin\Api\ImojeApiInterface;
+use BitBag\SyliusImojePlugin\Resolver\SignatureResolverInterface;
+use Payum\Core\Bridge\Spl\ArrayObject;
+use Payum\Core\Reply\HttpPostRedirect;
+use Payum\Core\Request\Capture;
+use PhpSpec\ObjectBehavior;
+use Sylius\Bundle\PayumBundle\Model\PaymentSecurityTokenInterface;
+use Sylius\Component\Core\Model\AddressInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\PaymentInterface;
+use Sylius\Component\Customer\Model\CustomerInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+final class CaptureActionSpec extends ObjectBehavior
+{
+    public function let(
+        SignatureResolverInterface $signatureResolver,
+    ): void {
+        $this->beConstructedWith($signatureResolver);
+    }
+
+    public function it_should_return_true_when_request_and_model_is_valid(
+        Capture $request,
+        ArrayObject $arrayObject,
+    ): void {
+        $request->getModel()->willReturn($arrayObject);
+        $this->supports($request)->shouldReturn(true);
+    }
+
+    public function it_should_return_false_when_model_is_invalid(
+        Capture $request,
+    ): void {
+        $request->getModel()->willReturn(null);
+        $this->supports($request)->shouldReturn(false);
+    }
+
+    public function it_should_return_false_when_request_invalid(
+        Request $request,
+    ): void {
+        $this->supports($request)->shouldReturn(false);
+    }
```

Comment:
you should mock failed $request value here, otherwise we don't know why support will return false.